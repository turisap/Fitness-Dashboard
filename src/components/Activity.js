/**
 * Created by HP on 19-Jan-18.
 */
import React from 'react';
import { connect } from 'react-redux';
import { Card, Icon, Image } from 'semantic-ui-react';
import {CloseButton} from 'react-svg-buttons'

const Activity = props => {
    const {id, name, type, description, average_speed, max_speed, location_country} = props.activity;
    const src = `./assets/${type.toLowerCase()}.png`;
    return (
        <Card>
            <Image src={src} />
            <Card.Content>
                <Card.Header>{name}</Card.Header>
                <Card.Meta>{type}</Card.Meta>
                <Card.Description>Average_speed: {average_speed}, Max speed: {max_speed} </Card.Description>
            </Card.Content>
            <Card.Content extra>
                <Icon name='home' />
                {location_country}
            </Card.Content>
        </Card>
    )
};

export default Activity;