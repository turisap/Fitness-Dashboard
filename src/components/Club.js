/**
 * Created by HP on 21-Jan-18.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';
import { connect } from 'react-redux';
import classNames from 'classnames';

import {leaveClub} from '../actions/clubs';

const Club = props => {
    const {id, name, cover_photo_small, sport_type, city, country} = props.club;
    return (
        <div>
            <img src={cover_photo_small}/>
            <h4>{name}</h4>
            <p>{sport_type}</p>
            <p>{[city, ' ', country]}</p>
            <Icon name="delete" color="red" size="huge" onClick={() => props.leaveClub(id)}
                  className={classNames({'club-leave-icon--loading' : props.loadingLeaveButton})}/>
        </div>
    )
};

const mapDispatchToProps = dispatch => ({
    leaveClub : id => dispatch(leaveClub(id))
});

const mapStateToProps = state => ({
    loadingLeaveButton : state.menuElements.loadingElements.leavingClub
});

Club.proptypes = {
    club : PropTypes.object.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Club);