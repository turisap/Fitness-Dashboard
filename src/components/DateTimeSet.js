/**
 * Created by HP on 26-Dec-17.
 */
import React from 'react';
import {DatePicker, TimePicker} from 'antd';
import moment from 'moment';
import PropTypes from 'prop-types';


/**
 * Represents date and time of start and end on the create page
 * @param props
 * @constructor
 */
const DateTimeSet = props => {
    const {defaultDate, defaultTime, context, onChange} = props;
    return (
        <div className="date-time-set">
            <DatePicker onChange={onChange} defaultValue={defaultDate} disabledDate={(current, cont) => disabledDate(current, context)}/>
            <TimePicker onChange={onChange} defaultValue={defaultTime} format="HH:mm"/>
        </div>
    )
};

/**
 * Validation of props
 * @type {{defaultDate, defaultTime, disabledDate}}
 */
/*DateTimeSet.propTypes = {
    defaultDate  : PropTypes.instanceOf(moment).isRequired,
    defaultTime  : PropTypes.instanceOf(moment).isRequired,
    context      : PropTypes.string.isRequired,
};*/

/**
 * Disables days in datepicker based on context from props (beginning of disabling)
 * @param current
 * @param context
 * @returns {boolean|*}
 */
const disabledDate = (current, context) => {
    const comparison = (context === 'today') ? moment() : moment().add(1, 'day');
    return current.isBefore(comparison, 'day')
};


export default DateTimeSet;