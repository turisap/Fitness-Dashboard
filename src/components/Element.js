/**
 * Created by HP on 17-Jan-18.
 */
import React from 'react';
import classNames from 'classnames';
import { Input } from 'semantic-ui-react';
import { connect } from 'react-redux';

import Validator from '../validators/InputDataValidator';

const Element = props => {
    return (
        <div className={classNames('dashboard-element', {'square' : props.square})} style={{border: 'red 1px solid'}}>
            <p>{props.title}</p>
            <p>{props.subtitle}</p>
            {(props.changeable && props.type === 'weight') &&
            <Input loading={props.loadingWeight} placeholder='Have lost sth?' action={{content: 'Change', onClick : changeMyWeight}} />}
        </div>
    )
};


const changeMyWeight = (e, data) => {
    console.log()
};

const mapStateToProps = state => ({
    loadingWeight : state.menuElements.loadingChangingWeight
});

export default connect(mapStateToProps, undefined)(Element);