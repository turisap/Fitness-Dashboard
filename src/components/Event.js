/**
 * Created by HP on 25-Dec-17.
 */
import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const Event = props => {
    const {is_free, start, name, logo, description, end, id} = props.event;

    return (
        <div className="event card mb-3">
            <Link className="event__link" to={`/event/${id}`}>
                <div className="event__picture-container card-body">
                    <img src={logo && logo.url} className="event__picture"/>
                    {is_free && <div className="event__price-type">FREE</div>}
                </div>
                <div className="event__info-container card-body">
                    <div className="event__date">{start && start.local}</div>
                    <div className="event__title">{name && name.text}</div>
                    <div className="event__description">{description.text ? description.text.substring(0, 100) + '..' : ''}</div>
                </div>
                <div className="event__footer card-footer">
                    <div className="event__tags">{end && end.timezone}</div>
                    <div className="event__save"></div>
                </div>
            </Link>
        </div>
    )
};

/**
 * Validating passing of event as a prop
 * @type {{event}}
 */
Event.propTypes = {
    event : PropTypes.object.isRequired,
};


export default Event;