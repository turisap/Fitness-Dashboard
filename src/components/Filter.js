/**
 * Created by HP on 30-Dec-17.
 */
import React from 'react';
import PropTypes from 'prop-types';


const Filter = props => (
    <div className='filter-bar__filter-element'
        onClick={props.onClick}>{props.title}
        <span className="filter-bar__remove">x</span>
    </div>
);

Filter.proptypes = {
    onClick: PropTypes.func.isRequired,
    title  : PropTypes.string.isRequired,
    show   : PropTypes.bool.isRequired,
};

export default Filter;