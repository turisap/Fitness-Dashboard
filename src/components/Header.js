/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {toggleNavbar} from '../actions/userData';
import classNames from 'classnames';

export const Header = (props) => (
    <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light navbar-right">
            <NavLink className="navbar-brand" to="/">Revent</NavLink>
            <button className="navbar-toggler" type="button" onClick={props.toggleNavbar}>
                <span className="navbar-toggler-icon"></span>
            </button>
            <div className={classNames('collapse', 'navbar-collapse', {'clicked' : props.clicked})} id="navbarColor03">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/browse">Browse</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/create">Create</NavLink>
                    </li>
                    {props.authenticated
                    ?
                        <li className="nav-item">
                            <NavLink activeClassName="active" className="nav-link" to="/signout">Sign OUT</NavLink>
                        </li>
                    :
                        [
                            <li className="nav-item" key="1">
                                <NavLink activeClassName="active" className="nav-link" to="/signup">Sign Up</NavLink>
                            </li>,
                            <li className="nav-item" key="2">
                                <NavLink activeClassName="active" className="nav-link" to="/signin">Sign In</NavLink>
                            </li>
                        ]
                    }
                </ul>
            </div>
        </nav>
    </div>
);


const mapStateToProps = (state) => ({
    authenticated : state.userData.authenticated,
    clicked : state.userData.clicked
});

const mapDispatchToProps = dispatch => ({
    toggleNavbar : () => dispatch(toggleNavbar())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);