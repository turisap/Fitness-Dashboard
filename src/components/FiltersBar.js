/**
 * Created by HP on 25-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';



import {removeFilter} from '../actions/filters';
import Filter from './Filter';
import { CSSTransitionGroup } from 'react-transition-group';


/**
 * Represents filter bar on the browse events page
 * @param props
 * @constructor
 */
export const FiltersBar = props => (
    <div className="filter-bar">
        <CSSTransitionGroup
            transitionName="filter"
            transitionEnterTimeout={500}
            transitionLeaveTimeout={500}>
            {filters({...props.filters}, props.removeFilter)}
        </CSSTransitionGroup>
    </div>
);


/**
 * Loop for getting divs with names of filters
 * @param list
 * @param dispatch
 * @returns {Array}
 */
const filters = (list, dispatch) => {
    delete list.allCategories;
    delete list.category;
    delete list.startRange;
    delete list.endRange;
    const filters = [];
    for (const filter in list){
        if (list[filter]){
            filters.push(
                <Filter
                    show={list[filter]}
                    key={filter}
                    title={list[filter]}
                    onClick={() => dispatch(removeFilter(filter))}/>
                    )
        }
    }
    return filters;
};


/**
 * Validation of props
 * @type {{filters}}
 */
FiltersBar.propType = {
    filters : PropTypes.object.isRequired
};



/**
 * Mapping filters object from Redux store
 * @param state
 */
const mapStateToProps = state => ({
    filters : state.filters,
});



/**
 * Mapping removing filters actions to props
 * @param dispatch
 */
const mapDispatchToProps = dispatch => ({
    removeFilter : filter => dispatch(removeFilter(filter))
});

export default connect(mapStateToProps, mapDispatchToProps)(FiltersBar);