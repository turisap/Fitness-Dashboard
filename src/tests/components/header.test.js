/**
 * Created by HP on 08-Jan-18.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import {Header} from '../../components/Header';

test('should render the header component properly', () => {
   const wraper = shallow(<Header/>);
   expect(wraper).toMatchSnapshot();
});

test('should call toggleNavbar()', () => {
    const toggleNavbar = jest.fn();
    const wrapper = shallow(<Header toggleNavbar={toggleNavbar}/>);
    wrapper.find('.navbar-toggler').simulate('click');
    expect(toggleNavbar).toBeCalled();
});