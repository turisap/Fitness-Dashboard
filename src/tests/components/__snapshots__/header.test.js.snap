// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header component properly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light navbar-right"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      Revent
    </NavLink>
    <button
      className="navbar-toggler"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarColor03"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/browse"
          >
            Browse
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/create"
          >
            Create
          </NavLink>
        </li>
        <li
          className="nav-item"
          key="1"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/signup"
          >
            Sign Up
          </NavLink>
        </li>
        <li
          className="nav-item"
          key="2"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/signin"
          >
            Sign In
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
